src/js/flows/executeTableSearch.js

```diff
@@ -28,10 +29,10 @@ export default function executeTableSearch(
       )
       .stats((stats) => dispatch(Viewer.setStats(tabId, stats)))
       .error((error) => dispatch(Notice.set(ErrorFactory.create(error))))
-      .end((_id, count) =>
+      .end((_id, count) => {
+        resolve()
```

Comment:
Yesterday I was thinking that this would be the way to do it, and I see you've found it too. But now I think we could make this simpler by having `executeTableSearch` return the promise. (I noticed we don't rely on the return value of `executeTableSearch` anywhere). Then in submitSearch(), have the last line be `return executeTableSearch(...)`. And `fetchNextPage()` wouldn't need to change at all.

For example:


```js
function executeTableSearch(...):  {
  return function(dispatch) {
    return new Promise((resolve, reject) => {
        ....
        .error((error) => {
          dispatch(Notice.set(ErrorFactory.create(error)))
          reject(error)
        })
        .end((_id, count) => {
          dispatch(Viewer.setEndStatus(tabId, endStatus(count)))
          resolve()
        })
      ...
    })
  }
}
```

And the switch statement at the end of submitSearch would now look like:

```js
switch (args.type) {
      case "analytics":
        return dispatch(executeTableSearch(tabId, args))
      case "zoom":
        return dispatch(executeTableSearch(tabId, args))
      default:
        dispatch(executeHistogramSearch(tabId, args, prevArgs))
        return dispatch(executeTableSearch(tabId, args))
    }
```

In the case where we bail early if the syntax isn't valid (line 25 of submitSearch.js), it might be good to return `Promise.reject()` to keep the return values the same type.

```js
if (!dispatch(SearchBar.validate())) return Promise.reject()
```

Reply:
Good suggestion. I think I've got this sorted now in the last commit. Reverted the changes in `fetchNextPage` as well.